tools/gen-spec-dependencies

```diff
@@ -21,19 +21,35 @@
 
 import sys
 import os
+import re
 import fileinput
 import subprocess
+import json
 
-min_base_version_path = os.path.join(os.path.dirname(os.path.realpath(sys.argv[0])), "min-base-version")
+tools_dir = os.path.dirname(os.path.realpath(sys.argv[0]))
+min_base_version_path = os.path.join(tools_dir, "min-base-version")
+package_json = os.path.join(os.path.dirname(tools_dir), "package.json")
 
 if len(sys.argv) != 2:
     sys.stderr.write("Usage: %s <specfile>\n" % sys.argv[0])
     sys.exit(1)
 
+# for resolving %{npm-version-*}
+with open(package_json) as f:
+    dependencies = json.load(f)["dependencies"]
+
 curpkg = None
 pkg_base_dep = None
+npm_version_re = re.compile(r"%{npm-version-(.*?)}")
+
 for line in fileinput.input(sys.argv[1], inplace=True):
-    if line.startswith("%package"):
+    npm_version_match = npm_version_re.search(line)
```

Comment:
Do we need to worry about multiple occurrences of the pattern in a single line?  I'd say no.

Reply:
I don't think so; every dependency needs to be on a separate line, and I can't think of a case where you need the version of a different NPM module.